### PR TITLE
Trill line playback

### PR DESCRIPTION
**Pull request by jimka2001**
This branch implements the playback for line type trills

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
Hmm, strange Travis failure:
/home/travis/build/musescore/MuseScore/build.debug/mtest/libmscore/layout/tst_benchmark: malloc(): memory corruption (fast): 0x000000000e1bcae0

**Comment by jimka2001** [comment]
Jojo, I didn't want to make the INVALID value -1 because I didn't think it would work with an unsigned char.  And I also didn't want to make it the first one, because as I've seen some places in muesescore xml-out/in depend on the integer values of certain enums.  I didn't want to change the semantics of integer values of Trill::Type if they happen to appear in an old xml file.

**Comment by Jojo-Schmitz** [comment]
adding it at the start of the enum and setting it to -1 does not change any of the other values. But indeed that char needs to get changed to signed char, this should not cause any problems though, except for the odd warung reg. some printf format specifier, which can get fixed quite easily

**Comment by Jojo-Schmitz** [comment]
playArticulation has just been replaced by just play, I guess that is what's now causing the merge conflict hiere.

**Comment by jimka2001** [comment]
Did someone replace playArticulation with play?

**Comment by lasconic** [comment]
Yes I did. It's not nicely harmonized with note, glissando, trill and bend.

**Comment by jimka2001** [comment]
Hi Nicolas, can you point me to the commit where you changed playArticulation to play?

**Comment by jimka2001** [comment]
This is my first attempt at bringing this branch up to date with the latest master. Something unexpected has changed in the way trills are playing now.  I will be looking into it.

**Comment by jimka2001** [comment]
Not sure why this test failure is occurring.  It thinks there is an occurrence of PLAY_ARTICULATILON, even though commit b395cdb removed PLAY_ARTICULATILON.

**Comment by jimka2001** [comment]
Trying again in case Travis is just confused

**Comment by MarcSabatella** [comment]
To find the change responsible for any given line of code, view the file on GitHub and press the Blamr button.  Terrible name, but invaluable function.

**Comment by lasconic** [comment]
Here is the commit https://github.com/musescore/MuseScore/commit/f6959ace914fce63bbaa3bfb0a090bea9cd742ea

**Comment by jimka2001** [comment]
Hi Nicolas, wrt f6959ac I notice that you didn't change the accessor function playArticulation, setPlayArticulation, playGlissando, and setPlayGlissando names.   Was that intentional?

**Comment by lasconic** [comment]
I was more worry about consistency in the file format. So I'm fine with "playGlissando" or "play" in the code.

**Comment by jimka2001** [comment]
This commit fixes a significant aesthetic musical problem compared to the previous version, in that it makes the ornaments sound much more "regular", and less "jumpy".

**Comment by jimka2001** [comment]
@lasconic I have endeavored to obey the coding guidelines with respect to banner style indentation and parens around logical expressions.   I didn't know exactly how to handle redundant braces {} in single line while loops. 

I'm not particularly fond (personally) of the redundant conditionals, but I resisted refactoring this logic into local functions, taking people's aversion thereto into account.  I hope the code is readable.

**Comment by jimka2001** [comment]
ready to merge in unless anyone has objections.

File: libmscore/rendermidi.cpp [hunk 1]
```diff
@@ -1052,9 +1080,118 @@ void renderNoteArticulation(NoteEventList* events, Note * note, bool chromatic,
...
+            }
+            return false;
+      }
+      
+//---------------------------------------------------------
+//   renderNoteArticulation
+//---------------------------------------------------------
+bool renderNoteArticulation(NoteEventList* events, Note * note, bool chromatic, Trill::Type trillType, MScore::OrnamentStyle ornamentStyle)
+      {
+      map<Trill::Type,ArticulationType> articulationMap = {
```
Comment (jimka2001) [inline libmscore/rendermidi.cpp]: this is the current list of line articulations supported.

**Pull request merged**